task/task.go

```diff
@@ -44,11 +44,9 @@ func New(ctx context.Context, cloud common.Cloud, identifier common.Identifier,
 	}
 }
 
+// Task defines the interface implemented by provider-specific task resources.
 type Task interface {
-	Read(ctx context.Context) error
-
-	Create(ctx context.Context) error
-	Delete(ctx context.Context) error
+	common.Resource
```

Comment:
The kind of thing I'd prefer to see on a separate pull request but am still thankful for. 😍